### PR TITLE
First stab at a reference section for the schema definition

### PR DESCRIPTION
**Pull request by draxil**
Okay, I've had a stab at writing a reference section to address #18 .

I think my examples of transformers may be a bit cumbersome and I did wonder if maybe I should split this into a separate POD file. Let me know what you think.

Cheers.

### CODE REVIEW

File: t/018_reference_examples.t [hunk 1]
```diff
@@ -0,0 +1,235 @@
+use strict;
+use lib 'lib';
+use Test::More;
+use Data::Processor;
+
+
+subtest 'members simple' => sub{
+
+    my $schema = {
+        cooridinates => {
```
Comment (maettu) [inline t/018_reference_examples.t]: possible typo "coordinates"?

**Comment by maettu** [comment]
I believe that you as a user might know better how the documentation should look like than me, the implementer. :-)

So, thank you for this contribution.

Please feel free to organise the POD in way that is sensible to you. If you move content to a second POD, please insert a link into the main document.
On CPAN, I like to read the documentation with as many clicks as possible. E.g. after searching a module on metacpan, I would like to get to immediately get to the docu page. (I don't like too much having to search the "Tutorial" file, or click a link on the main documentation page to get there.)
It might make sense to break out the transformer doc, however. I wonder if that feature isn't a bit edgy, anyway..

One small detail: In the array explanation, consider using plurals. "houses" and "windows" seem more natural to me for things that are "many". I often do "for my $house (@houses)" etc.

**Comment by draxil** [comment]
Okay further modified it to fix the typo and change the array examples as you suggested.

I haven't split anything out, on balance it didn't quite seem worth it. But I suppose it could be a good candidate should the documentation start to get unwieldy in future.

**Comment by draxil** [comment]
Thanks.

I don't think I have anything further on this front.

**Comment by maettu** [comment]
uploaded to CPAN. 
Also, I decided that with your documentation the module should now be version 1.0.0 :-)

**Comment by draxil** [comment]
Excellent :)

Thanks!

**Pull request merged**